rclpy/rclpy/executors.py

```diff
@@ -118,14 +117,20 @@ def wait(self, timeout_sec: Optional[float] = None) -> bool:
         return True
 
 
-async def await_or_execute(callback: FunctionOrCoroutineFunction[T], *args: Any) -> T:
+@overload
+async def await_or_execute(callback: Callable[..., Coroutine[Any, Any, T]], *args: Any) -> T: ...
+
+
+@overload
+async def await_or_execute(callback: Callable[..., T], *args: Any) -> T: ...
+
+
+async def await_or_execute(callback: Callable[..., Any], *args: Any) -> Any:
```

Comment:
Today I learned! https://docs.python.org/3/library/typing.html#typing.overload

I do have a question: the `@overload`s above means that this does not actually take in a `Callable` that returns `Any`, right? They _must_ return either a `Coroutine[Any, Any, T]` or `T`, right?

Reply:
I might be misunderstanding but, someone could pass in a function which returns Any which would update the return type of await_or_execute to return Any. To stop that the user would have to run a type checker with --disallow-any on their code.

Reply:
Yeah, sorry, I meant: This works fine at runtime, but mypy would complain about a call to `await_or_execute()` that does not use a `Callable` of return type `Coroutine[Any, Any, T]` or `T`, correct?

Reply:
Yeah.